### DISCUSSION TITLE
containers won't start after manual backup removal

### DISCUSSION THREAD

**Discussion by MariusQuabeck** [nextcloud/all-in-one#2350] [Questions] [answered]
I regularly have to remove old backups to free up disk space.
This usually leads to my containers not starting up again until I manually remove the database lock file.

db container:
`Waiting for backup container to finish...
If this is incorrect because the backup container is not running anymore (because it was forcefully killed), you might delete the lock file:
sudo docker exec --user root nextcloud-aio-database rm /mnt/data/backup-is-running`

I assume this file is being placed there when I'm trying to verify backup integrity to remove old entries in the list, as stated by the readme section about backup removal.

backup container:
`Performing backup...
Starting the backup...
Creating archive at "/mnt/borgbackup/borg::20230404_160614-nextcloud-aio"
Synchronizing chunks cache...
Archives: 0, w/ cached Idx: 0, w/ outdated Idx: 0, w/o cached Idx: 0.
Done.
`

steps to reproduce:
create a backup from AIO page, remove backup as stated in readme section, try emptying the backup list on AIO page by verifying, get stuck and manually reboot host or restart containers, database not starting because of lock file, remove lock file

**Comment by MariusQuabeck** [comment]
also subsequently backups break
`Performing backup...
Starting the backup...
Failed to create/acquire the lock /root/.cache/borg/5bb3af51325768c0c5779cb3349f08b520a324d67e5de1d5f35b21bb813f1e16/lock.exclusive (timeout).
Deleting the failed backup archive...
Failed to create/acquire the lock /root/.cache/borg/5bb3af51325768c0c5779cb3349f08b520a324d67e5de1d5f35b21bb813f1e16/lock.exclusive (timeout).
Backup failed!`

**Comment by szaimen** [comment]
Hi, thanks for the report!

> I regularly have to remove old backups to free up disk space.

Did you know that this is done automatically? See https://github.com/nextcloud/all-in-one/discussions/1675

> I assume this file is being placed there when I'm trying to verify backup

~~Indeed and we are going to improve this as the lock is not really needed in the case of verifying a backup.~~

Actually it looks like the logic only places such a file in there in case of backup or restore:
![image](https://user-images.githubusercontent.com/42591237/230070440-1ff64c4b-9100-483c-9246-f1d41c85757a.png)

Is it possible that you rebooted your server while a backup or restore was running? That would make it stuck.

> get stuck and manually reboot host or restart containers, database not starting because of lock file

I don't think that it gets stuck, the integrity check just can take up to a few hours depending on the backup archive. But it should also not prevent the database from starting, see above. However rebooting the machine in any case actually can make it get stuck.

> also subsequently backups break `Performing backup... Starting the backup... Failed to create/acquire the lock /root/.cache/borg/5bb3af51325768c0c5779cb3349f08b520a324d67e5de1d5f35b21bb813f1e16/lock.exclusive (timeout). Deleting the failed backup archive... Failed to create/acquire the lock /root/.cache/borg/5bb3af51325768c0c5779cb3349f08b520a324d67e5de1d5f35b21bb813f1e16/lock.exclusive (timeout). Backup failed!`

I fear you will need to run `sudo borg break-lock /mnt/backup/borg` then since you basically interrupted borg from doing its thing which borg doesnt really like. See https://borgbackup.readthedocs.io/en/stable/usage/lock.html#borg-break-lock

**Reply by Fuseteam** [reply]
would this be with `docker exec nextcloud-aio-borgbackup sudo borg break-lock /mnt/backup/borg`?

**Reply by Fuseteam** [reply]
ah found this: https://github.com/nextcloud/all-in-one/discussions/2069#discussioncomment-5120090

**Reply by Fuseteam** [reply]
and it worked :tada:

**Comment by MariusQuabeck** [comment]
Hmm for me it never removed old backups, recently even Nextcloud Office stopped working because it run out of disk space, that's why I manually removed old backups.

I had to reboot since the backup verifying was stuck after removing old backups.

I did run the break-lock command but backups still wont work
`Performing backup...
Starting the backup...
Failed to create/acquire the lock /root/.cache/borg/5bb3af51325768c0c5779cb3349f08b520a324d67e5de1d5f35b21bb813f1e16/lock.exclusive (timeout).
Deleting the failed backup archive...
Failed to create/acquire the lock /root/.cache/borg/5bb3af51325768c0c5779cb3349f08b520a324d67e5de1d5f35b21bb813f1e16/lock.exclusive (timeout).
Backup failed!
`

**Comment by szaimen** [comment]
Maybe I wasnt clear enough. Only creating a new backup successfully will remove old ones based on the linked prune policy. So if you are already out of space it will not be able to prune old ones.

**Comment by MariusQuabeck** [comment]
yea that is what I mean, it created daily backups until it ran out of disk space and Nextcloud Office started complaining which is why I then had to remove backups manually.
So maybe it does work but the quota/rentention policy needs adjusting, I'll have an eye on that when I get backups to work again, at the moment it fails with the above mentioned message in the log.

**Comment by szaimen** [comment]
> I did run the break-lock command but backups still wont work
> `Performing backup... Starting the backup... Failed to create/acquire the lock /root/.cache/borg/5bb3af51325768c0c5779cb3349f08b520a324d67e5de1d5f35b21bb813f1e16/lock.exclusive (timeout). Deleting the failed backup archive... Failed to create/acquire the lock /root/.cache/borg/5bb3af51325768c0c5779cb3349f08b520a324d67e5de1d5f35b21bb813f1e16/lock.exclusive (timeout). Backup failed! `

Regarding that issue I am not completely sure how to resolve that ad it seems that possibly the out of space issue created issues with the borg cache. You could try to clear that by clearing the borg cache, stored in /var/lib/docker/volumes/nextcloud_aio_backup_cache/_data but I never tried that and it is possible that you would break things more by doing so.

**Comment by szaimen** [comment]
> yea that is what I mean, it created daily backups until it ran out of disk space and Nextcloud Office started complaining which is why I then had to remove backups manually.
> So maybe it does work but the quota/rentention policy needs adjusting,

I guess the best advice would be moving the borg repository to a different drive then.

**Comment by MariusQuabeck** [comment]
thank you, I'll give this a try over the weekend

**Comment by szaimen** [comment, marked as answer]
Hi Marius, did it work? :)

**Reply by MariusQuabeck** [reply]
unfortunately no. I could not find any instructions online for how to clear the borg cache.
Is there a way to switch out the backup container for a working one? I have backups enabled on linode for now but thats expensive :D

**Comment by szaimen** [comment]
This should clear the borg cache:

```
sudo docker rm nextcloud-aio-borgbackup 
sudo docker volume rm nextcloud_aio_backup_cache
```

**Reply by MariusQuabeck** [reply]
I've just managed to initialise a new backup and it seems to be working now!
I'll be trying a restore next, if it doesn't work then I will try what you just wrote.

Appreciate your patience, thanks Simon!

**Comment by MariusQuabeck** [comment]
success!

```
Performing backup...
Starting the backup...
Creating archive at "/mnt/borgbackup/borg::20230427_141728-nextcloud-aio"
------------------------------------------------------------------------------
Repository: /mnt/borgbackup/borg
Archive name: 20230427_141728-nextcloud-aio
Archive fingerprint: ee4dece9520ffb3823de35c41863b1f6ed83a468b3d963feb01a8eac2aa01115
Time (start): Thu, 2023-04-27 14:17:29
Time (end):   Thu, 2023-04-27 14:23:01
Duration: 5 minutes 32.85 seconds
Number of files: 51739
Utilization of max. archive size: 0%
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
This archive:               24.96 GB             18.26 GB             18.04 GB
All archives:               24.96 GB             18.26 GB             18.05 GB

                       Unique chunks         Total chunks
Chunk index:                   53637                59451
------------------------------------------------------------------------------
Pruning the archives...
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
Deleted data:                    0 B                  0 B                  0 B
All archives:               24.96 GB             18.26 GB             18.05 GB

                       Unique chunks         Total chunks
Chunk index:                   53637                59451
------------------------------------------------------------------------------
Compacting the archives...
Backup finished successfully on 27.04.2023 - 14:23:04 (00 hours 05 minutes 36 seconds)

```

**Reply by szaimen** [reply]
Great! :)
